### PR TITLE
Dism more data still

### PR DESCRIPTION
**Pull request by PikalaxALT**
Also decomp Hall of Fame because I have no self-control.

### CODE REVIEW

File: src/hall_of_fame.c [hunk 1]
```diff
@@ -0,0 +1,1293 @@
...
+static void Hof_WaitBorderFadeAway(u8 taskId)
+{
+    if (gTasks[taskId].data[3] > 15)
+    {
+        gTasks[taskId].func = Hof_SpawnPlayerPic;
+    }
+    else
+    {
+        gTasks[taskId].data[3]++;
+        SetGpuReg(REG_OFFSET_BLDALPHA, 256 * gTasks[taskId].data[3]);
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: use `BLDALPHA_BLEND` macro

File: src/hall_of_fame.c [hunk 2]
```diff
@@ -0,0 +1,1293 @@
...
+    }
+}
+
+static void Hof_SpawnPlayerPic(u8 taskId)
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: Task_Hof_SpawnPlayerPic

File: src/hall_of_fame.c [hunk 3]
```diff
@@ -0,0 +1,1293 @@
...
+    {
+        if (currMon->species != 0)
+            gTasks[taskId].data[4]++;
+    }
+
+    currMon = &savedTeams->mon[0];
+
+    for (i = 0; i < PARTY_SIZE; i++, currMon++)
+    {
+        if (currMon->species != 0)
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: `SPECIES_NONE`

File: src/hall_of_fame.c [hunk 4]
```diff
@@ -0,0 +1,1293 @@
...
+        savedTeams++;
+
+    currMon = &savedTeams->mon[0];
+    sSelectedPaletteIndices = 0;
+    gTasks[taskId].data[2] = 0;
+    gTasks[taskId].data[4] = 0;
+
+    for (i = 0; i < PARTY_SIZE; i++, currMon++)
+    {
+        if (currMon->species != 0)
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: `SPECIES_NONE`

File: src/hall_of_fame.c [hunk 5]
```diff
@@ -0,0 +1,1293 @@
...
+
+    gTasks[taskId].func = Task_HofPC_PrintMonInfo;
+}
+
+static void Task_HofPC_PrintMonInfo(u8 taskId)
+{
+    struct HallofFameTeam* savedTeams = sHofMonPtr;
+    struct HallofFameMon* currMon;
+    u16 i;
+    u16 currMonID;
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: Is there a reason to use both ID and Id?

File: src/hall_of_fame.c [hunk 6]
```diff
@@ -0,0 +1,1293 @@
...
+        rand = (Random() % 4) + 8;
+        sprite->pos2.x = rand * gSineTable[tableID] / 256;
+
+        sprite->data[0] += 4;
+    }
+}
+
+static bool8 Hof_SpawnConfetti(void)
+{
+    u8 spriteID;
```
Comment (jiangzhengwenjz) [inline src/hall_of_fame.c]: Is there a reason to use both ID and Id?

File: src/script.c [hunk 7]
```diff
@@ -546,7 +546,7 @@ u8 *sub_8069E48(void)
     }
 }
 
-void sub_8069EA4(u8 *script, u16 scriptSize)
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/script.c]: Not necessarily.

File: data/mevent/script_ish.inc [hunk 8]
```diff
@@ -1,64 +1,64 @@
-gUnknown_84687E0:: @ 84687E0
+gMEventClientScript_IntialListen:: @ 84687E0
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply data/mevent/script_ish.inc]: My fingers are too fast for my keyboard

File: data/mevent/script_ish.inc [hunk 9]
```diff
@@ -1,64 +1,64 @@
...
-gUnknown_84688B8::
+gMEventClientScript_AskWouldLikeToTossCard::
 	.4byte 0x0d, 0x00 @ REQUEST PARAMETER
 	.4byte 0x0e, 0x00 @ SEND PARAMETER
 	.4byte 0x03, 0x00 @ WAIT SEND
 	.4byte 0x02, 0x10 @ RECEIVE ID(0x10)
 	.4byte 0x04, 0x00 @ JUMP TO RECEIVE BUFFER
 
-gUnknown_84688E0::
+gMEventClientScript_OtherTrainerCanceled:: @ comm cancelled
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply data/mevent/script_ish.inc]: I'll go with one `l` as that seems to be the dominant variant in the game's text.

File: include/global.h [hunk 10]
```diff
@@ -690,11 +690,11 @@ struct MEventBuffer_3430
...
 {
-    /*0x000 0x3120*/ struct MEventBuffer_3120 buffer_000;
-    /*0x1c0 0x32e0*/ struct MEventBuffer_32E0 buffer_1c0;
+    /*0x000 0x3120*/ struct MEWonderNewsStruct menews;
+    /*0x1c0 0x32e0*/ struct MEWonderCardStruct mecard;
     /*0x310 0x3430*/ struct MEventBuffer_3430 buffer_310;
-    /*0x338 0x3458*/ u16 unk_338[4];
-    /*0x340 0x3460*/ struct MENewsJisanStruct unk_340;
+    /*0x338 0x3458*/ u16 ec_profile_maybe[4];
+    /*0x340 0x3460*/ struct MENewsJisanStruct mejisan;
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/global.h]: i'll use snake_case for this actually

File: include/mevent_server.h [hunk 11]
```diff
@@ -46,12 +46,12 @@ struct mevent_client
     u32 cmdidx;
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/mevent_server.h]: nah

File: src/mevent_client.c [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+{
+    u32 result;
+    if (s_mevent_client_ptr == NULL)
+        return 6;
+    result = mevent_client_exec(s_mevent_client_ptr);
+    if (result == 6)
+    {
+        *a0 = s_mevent_client_ptr->param;
+        mevent_client_free_resources(s_mevent_client_ptr);
+        Free(s_mevent_client_ptr);
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/mevent_client.c]: nah

File: src/hall_of_fame.c [hunk 13]
```diff
@@ -0,0 +1,1293 @@
...
+    u32 tid;
+    u32 personality;
+    u16 species:9;
+    u16 lvl:7;
+    u8 nick[10];
+};
+
+struct HallofFameTeam
+{
+    struct HallofFameMon mon[PARTY_SIZE];
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/hall_of_fame.c]: nope

File: src/hall_of_fame.c [hunk 14]
```diff
@@ -0,0 +1,1293 @@
...
+static void Task_HofPC_CopySaveData(u8 taskId);
+static void Task_HofPC_DrawSpritesPrintText(u8 taskId);
+static void Task_HofPC_PrintMonInfo(u8 taskId);
+static void Task_HofPC_HandleInput(u8 taskId);
+static void Task_HofPC_HandlePaletteOnExit(u8 taskId);
+static void Task_HofPC_HandleExit(u8 taskId);
+static void Task_HofPC_PrintDataIsCorrupted(u8 taskId);
+static void Task_HofPC_ExitOnButtonPress(u8 taskId);
+static void HallOfFame_PrintWelcomeText(u8 a0, u8 a1);
+static void HallOfFame_PrintMonInfo(struct HallofFameMon * hofMon, u8 a1, u8 a2);
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/hall_of_fame.c]: the const qualifier is only needed when passing rodata, which never happens here

File: src/hall_of_fame.c [hunk 15]
```diff
@@ -0,0 +1,1293 @@
...
+            PlayCry1(currMon->species, 0);
+        HallOfFame_PrintMonInfo(currMon, 0, 14);
+        gTasks[taskId].data[3] = 120;
+        gTasks[taskId].func = Task_Hof_TryDisplayAnotherMon;
+    }
+}
+
+static void Task_Hof_TryDisplayAnotherMon(u8 taskId)
+{
+    u16 currPokeID = gTasks[taskId].data[1];
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/hall_of_fame.c]: fastest fingers in the ~~west~~ northeast

File: src/hall_of_fame.c (hunk 1 above)
Comment (PikalaxALT) [reply src/hall_of_fame.c]: doesn't match

File: src/mevent_client.c (hunk 12 above)
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/mevent_client.c]: nah

**Pull request merged**